haystack/schema.py

```diff
@@ -1045,3 +1056,31 @@ def load(cls, load_dir: Union[str, Path]):
         node_results = {file.stem: pd.read_csv(file, header=0, converters=converters) for file in csv_files}
         result = cls(node_results)
         return result
+
+
+class EvaluationDataset:
+    def __init__(self, name: str, labels: List[MultiLabel], meta: Dict[str, Any] = {}) -> None:
+        """
+        Set of labels belonging together and forming a well-specified and referenceable evaluation dataset.
+        """
+        self.name = name
+        self.labels = labels
+        self.meta = meta
+
+    def __len__(self) -> int:
+        return len(self.labels)
+
+
+class Corpus:
```

Comment:
As discussed in the call with Tanay, this looks like a FileStore maybe?

Reply:
Maybe `FileCorpus` so that users easily understand that this class groups files?